hadoop/src/main/java/org/apache/iotdb/tsfile/hadoop/TSFRecordReader.java

```diff
@@ -216,12 +230,12 @@ public float getProgress() throws IOException, InterruptedException {
 
   @Override
   public void close() throws IOException {
-    dataSet = null;
-    hdfsInputStream.close();
+    dataSetList.forEach(queryDataSet -> queryDataSet = null);
+    deviceIdList.forEach(deviceId -> deviceId = null);
```

Comment:
What is the point of setting the content to null instead of calling clear() or just set the fields to null?